assets/js/pages/SettingsPage/SettingsPage.jsx

```diff
@@ -24,28 +25,52 @@ import {
   getSoftwareUpdatesSettingsErrors,
 } from '@state/selectors/softwareUpdatesSettings';
 
+import { dismissNotification } from '@state/notifications';
+import { API_KEY_EXPIRATION_NOTIFICATION_ID } from '@state/sagas/settings';
+
 import SuseManagerConfig from '@common/SuseManagerConfig';
+import { format, parseISO } from 'date-fns';
 
 function SettingsPage() {
   const dispatch = useDispatch();
 
   const [loading, setLoading] = useState(false);
   const [apiKey, setApiKey] = useState(null);
-  const [showApiKey, setShowApiKey] = useState(false);
+  const [apiKeyExpiration, setApiKeyExpiration] = useState(null);
+  const [apiKeySettingModalOpen, setApiKeySettingsModalOpen] = useState(false);
   const [clearingSoftwareUpdatesSettings, setClearingSoftwareUpdatesSettings] =
     useState(false);
 
-  useEffect(() => {
-    setLoading(true);
-    get('/installation/api-key')
-      .then(({ data: { api_key: newApiKey } }) => {
-        apiKey !== undefined && setApiKey(newApiKey);
-        setLoading(false);
-      })
+  const fetchApiKeySettings = () =>
+    get('/settings/api_key')
+      .then(
+        ({ data: { generated_api_key: newApiKey, expire_at: expireAt } }) => {
+          setApiKey(newApiKey);
+          setApiKeyExpiration(expireAt);
+          setLoading(false);
+        }
+      )
       .catch((error) => {
         logError(error);
         setLoading(false);
       });
+
+  const saveApiKeySettings = (expiration) => {
+    setLoading(true);
+    patch('/settings/api_key', { expire_at: expiration }).then(
+      ({ data: { generated_api_key: newApiKey, expire_at: expireAt } }) => {
+        setApiKey(newApiKey);
+        setApiKeyExpiration(expireAt);
+        setLoading(false);
+        setApiKeySettingsModalOpen(false);
```

Comment:
Is this `setApiKeySettingsModalOpen` here correct?
I mean, when we click `Generate` button, we still need to continue in the modal, and show the new key/expiration date, etc
The modal should be closed only when we click `Close` as far as I know

Reply:
Nice catch, fixing rn